File: examples/preview.go
// Courtesy of rosettacode.org under the GNU Free Documentation License 1.2
// https://rosettacode.org/wiki/FizzBuzz#Go
// https://www.gnu.org/licenses/old-licenses/fdl-1.2.html

package main

import "fmt"

func main() {
	for i := 1; i <= 100; i++ {
		switch {
		case i%15 == 0:
			fmt.Println("FizzBuzz")
		case i%3 == 0:
			fmt.Println("Fizz")
		case i%5 == 0:
			fmt.Println("Buzz")
		default:
			fmt.Println(i)
		}
	}
}

File: examples/preview.rs
// Courtesy of rosettacode.org under the GNU Free Documentation License 1.2
// https://rosettacode.org/wiki/FizzBuzz#Rust
// https://www.gnu.org/licenses/old-licenses/fdl-1.2.html

use std::borrow::Cow; // Allows us to avoid unnecessary allocations
fn main() {
  (1..101).map(|n| match (n % 3, n % 5) {
    (0, 0) => "FizzBuzz".into(),
    (0, _) => "Fizz".into(),
    (_, 0) => "Buzz".into(),
    _ => Cow::from(n.to_string())
  }).for_each(|n| println!("{}", n));
}
